snap/hooks/post-refresh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Pre-refresh hook for charmed-mongodb snap
+
+set -eux
+
+export CONF="${SNAP_DATA}"/etc/mongod
+export ETC="${SNAP_DATA}"/etc/
+export MONGOS_CONFIG_FILE="${CONF}"/mongos.conf
+export LOGS="${SNAP_COMMON}"/var/log/mongodb
+export LDAP_CONF_DIR="${SNAP_DATA}"/etc/ldap/
+
+
+if [ ! -f $MONGOS_CONFIG_FILE ]; then
+    # Copy over the mongos.conf and create needed directories/files
+    echo "mongos configuration file does not exist."
+    echo "copying default config to ${MONGOS_CONFIG_FILE}"
+
+    # Grant permissions on $CONF
+    chown root:snap_daemon $CONF
+
+    cp -r ${SNAP}/etc/mongod.conf ${MONGOS_CONFIG_FILE}
```

Comment:
I think with this we're overriding everything the charm might have filled up until the refresh. 
we need to copy the previous revision of the `mongod.conf` file to ensure we're dealing with the most up-to-date content

Reply:
We are in the case where `mongos.conf` does not exist. In that case, we need to copy the default file from the squashfs to the `SNAP_DATA` directory.
The `-r` here is misguided and should be removed, I'll do it